pandas/util/testing.py

```diff
@@ -1074,8 +1074,12 @@ def assert_categorical_equal(left, right, check_dtype=True,
 def raise_assert_detail(obj, message, left, right, diff=None):
     if isinstance(left, np.ndarray):
         left = pprint_thing(left)
+    elif pd.api.types.is_categorical_dtype(left):
```

Comment:
Might be cleaner to import `is_categorical_dtype` from ` pandas.core.dtypes.common` at the top of the file, as other things are already being imported from it, and just use `is_categorical_dtype(left)` here.